flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -713,10 +713,18 @@ void CodeGenAction::generateLLVMIR() {
   pm.addPass(std::make_unique<Fortran::lower::VerifierPass>());
   pm.enableVerifier(/*verifyPasses=*/true);
 
+  const auto targetOpts = ci.getInvocation().getTargetOpts();
+  const llvm::Triple triple(targetOpts.triple);
+
+  MLIRToLLVMPassPipelineConfig config(level);
+
+  config.StackArrays = opts.StackArrays;
+  config.Underscoring = opts.Underscoring;
+  config.LoopVersioning = opts.LoopVersioning;
+  config.DebugInfo = opts.getDebugInfo();
+
```

Comment:
How about:

```suggestion
  MLIRToLLVMPassPipelineConfig config = {.OptLevel = level, .Underscoring = opts.Underscoring, .LoopVersioning = opts.LoopVersioning, .DebugInfo = opts.getDebugInfo()};
```

Reply:
I don't think that helps readability. I'd rather make a constructor that takes the opts, but I also [when this is in, and I'm doing the next step of adding another couple of fields] sometimes need to set some variables from language options, which complicates things. This is the only place where we're setting all of the options.

Reply:
Well, it's fewer characters as you would only type `config` once ;-) Also, it would discourage people from reconfiguring `config` later on (these things can be hard to track).

Whatever is your pref, this was just a suggestion.

Reply:
I'll do the stuff like the OffloadCnfig, then we can make the config const, so nobody [without const-casts and other obvious tricks] can accidentally change it later. It is a good point.